metadata_backend/api/operators.py

```diff
@@ -25,87 +25,123 @@ class BaseOperator(ABC):
     :param ABC: The abstract base class
     """
 
-    def __init__(self, db_type: str, content_type: str) -> None:
-        """Init needed variables, must be given by subclass."""
-        self.db_service = DBService(db_type)
+    def __init__(self, db_name: str, content_type: str,
+                 db_client: AsyncIOMotorClient) -> None:
+        """Init needed variables, must be given by subclass.
+
+        :param db_name: Name for database to save files to
+        :param content_type: Content type this operator handles (XML or JSON)
+        :param db_client: Motor client used for database connections. Should be
+        running on same loop with aiohttp, so needs to be passed from aiohttp
+        Application.
+        """
+        self.db_service = DBService(db_name, db_client)
         self.content_type = content_type
 
-    def read_metadata_object(self, schema_type: str, accession_id:
-                             str) -> Tuple[Union[Dict, str], str]:
+    async def create_metadata_object(self, schema_type: str,
+                                     data: Union[Dict, str]) -> str:
+        """Create new metadata object to database.
+
+        Data formatting and addition step for JSON or XML must be implemented
+        by corresponding subclass.
+
+        :param schema_type: Schema type of the object to read.
+        :param data: Data to be saved to database.
+        :returns: Accession id for the object inserted to database
+        """
+        accession_id = (await self.
+                        _format_data_to_create_and_add_to_db(schema_type,
+                                                             data))
+        LOG.info(f"Inserting file with schema {schema_type} to database "
+                 f"succeeded with accession id: {accession_id}")
```

Comment:
maybe this can be multiline with `"""` ?

Reply:
This is done to avoid adding linebreak, which multiline does.

Reply:
ok